common/src/main/java/org/dash/wallet/common/transactions/TransactionWrapperFactory.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.transactions
+
+import org.bitcoinj.core.Transaction
+
+interface TransactionWrapperFactory {
+    fun tryInclude(tx: Transaction): Pair<Boolean, TransactionWrapper?>
+}
```

Comment:
This wrapper factory will determine which wrapper should be used and return it along with the results of `tryInclude` from the wrapper.